course/models.py

```diff
@@ -0,0 +1,84 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+Category = [
+    ('BUSINESS_AND_MANAGEMENT', 'Business & Management'),
+    ('COMPUTER_SCIENCE_AND_IT', 'Computer Science & IT'),
+    ('CREATIVE_ARTS', 'Creative Arts'),
+    ('ECONOMICS', 'Economics'),
+    ('HISTORY', 'History'),
+    ('HEALTH_AND_PSYCHOLOGY', 'Health & Psychology'),
+    ('LITERATURE', 'Literature'),
+    ('LAW', 'Law'),
+    ('LANGUAGES_AND_CULTURES', 'Languages & Cultures'),
+    ('MATHS', 'Maths'),
+    ('MARKETING', 'Marketing'),
+    ('OTHER', 'Other'),
+    ('POLITICS_AND_SOCIETY', 'Politics & Society'),
+    ('SCIENCE_AND_ENGINEERING', 'Science & Engineering'),
+    ('SPORT_AND_LEISURE', 'Sport & Leisure'),
+]
+
+Level = [
+    ('B', 'Beginner'),
+    ('I', 'Intermediate'),
+    ('A', 'Advanced'),
+]
+
+
+class TeacherCoursesManager(models.Manager):
+
+    def in_price_range(self, min: int, max: int):
+        return self.filter(price__lte=max, price__gte=min)
+
+    def in_category(self, cat: str):
+        return self.filter(category=cat)
+
+    def got_experience(self, exp: int):
+        if exp > 5:
+            exp = 5
+        return self.filter(years_of_experience__gte=exp)
+
```

Comment:
We will also need a method in `TeacherCoursesManager` to search for teacher courses based on a keyword for the search view. Something like 
`self.filter(Q(course_name__icontains=keyword) | Q(description__icontains=keyword))`

using import `from django.db.models import Q`